### PR TITLE
Adding end-to-end test workflow and documentation

### PR DESCRIPTION
**Pull request by dkosteck**

### CODE REVIEW

**Review by jianzzha — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,4 +82,8 @@ All SR-IOV tests must be identified via either a tag or a commit hash, represent
...
+
+## Validating Changes
+
+All SR-IOV and common tests should be validated to work and, in cases where changes affect common code, that changes do not break existing tests or common code. To this end there are GitHub Actions running to validate formatting by way of a linter, a series of mock tests checking the common code, as well as an option to run tests on a lab machine. The linter and mocks run automatically on Pull Requests. The optional "real" testing can be run by applying the "e2e-test" label to the Pull Request. This will start an action which queues either an end-to-end test (where common code is affected) or runs the specific test modified (when no common code is affected). Because these tests take both time and resources, they should only be started after the PR is ready to merge as a final check of functionality. After the test run, the Action will remove the "e2e-test" label.
```
Comment (akiselev1) [inline CONTRIBUTING.md:89]: Should we mention here who is supposed/allowed to use e2e-test label? Formatting-wise, I suggest a new  paragraph starting at "The optional "real" testing...

**Review by akiselev1 — APPROVED** [review]
A couple of nitpicks on the documentation, otherwise looks good.

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    types: [labeled]
+jobs:
+  e2e-tests:
+    runs-on: self-hosted
+    if: contains(github.event.label.name, 'e2e-test')
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v3
```
Comment (akiselev1) [inline .github/workflows/e2e.yaml:14]: We should use setup-python@v4 to avoid "deprecated" warnings on every run.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dkosteck) [reply CONTRIBUTING.md]: @akiselev1 Addressed in the most recent commit

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (akiselev1) [reply .github/workflows/e2e.yaml:14]: I think updating setup-python version here is fine. I have another PR to update it in other actions. Python versions we can update later in a separate PR. There is no dependency on setup-python.

**Comment by akiselev1** [comment]
Definitely no matrix fro e2e tests. One python version is fine.

**Review by akiselev1 — APPROVED** [review]

**Pull request merged**